analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -89,9 +89,14 @@ private static void NamedSymbolAction(SymbolAnalysisContext context, HashSet<ISy
                 && new CSharpSymbolUsageCollector(context.Compilation, privateSymbols) is var usageCollector
                 && VisitDeclaringReferences(namedType, usageCollector, context.Compilation, includeGeneratedFile: true))
             {
-                var diagnostics = GetDiagnosticsForUnusedPrivateMembers(usageCollector, privateSymbols, SyntaxConstants.Private, fieldLikeSymbols)
-                                        .Concat(GetDiagnosticsForUsedButUnreadFields(usageCollector, privateSymbols));
-                context.ReportDiagnosticIfNonGenerated(diagnostics);
+                foreach (var diagnostic in DiagnosticsForUnusedPrivateMembers(usageCollector, privateSymbols, SyntaxConstants.Private, fieldLikeSymbols))
```

Comment:
I think it would be useful to have an overload that accepts a collection as before. We will avoid duplicating these `foreach` loops.